src/Workspaces/Core/Portable/Serialization/SerializerService.cs

```diff
@@ -4,67 +4,48 @@
 
 using System;
 using System.Collections.Concurrent;
-using System.Collections.Generic;
 using System.Collections.Immutable;
 using System.Composition;
-using System.Diagnostics;
 using System.Linq;
 using System.Runtime.Versioning;
 using System.Threading;
 using Microsoft.CodeAnalysis.Diagnostics;
 using Microsoft.CodeAnalysis.Host;
 using Microsoft.CodeAnalysis.Host.Mef;
 using Microsoft.CodeAnalysis.Internal.Log;
-using Microsoft.CodeAnalysis.PooledObjects;
 using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.Serialization;
 
 #if NETCOREAPP
 [SupportedOSPlatform("windows")]
 #endif
-internal partial class SerializerService : ISerializerService
+[method: Obsolete(MefConstruction.FactoryMethodMessage, error: true)]
+internal partial class SerializerService(SolutionServices workspaceServices) : ISerializerService
 {
     [ExportWorkspaceServiceFactory(typeof(ISerializerService), layer: ServiceLayer.Default), Shared]
-    internal sealed class Factory : IWorkspaceServiceFactory
+    [method: ImportingConstructor]
+    [method: Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+    internal sealed class Factory() : IWorkspaceServiceFactory
     {
-        [ImportingConstructor]
-        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
-        public Factory()
-        {
-        }
-
         [Obsolete(MefConstruction.FactoryMethodMessage, error: true)]
         public IWorkspaceService CreateService(HostWorkspaceServices workspaceServices)
             => new SerializerService(workspaceServices.SolutionServices);
     }
 
     private static readonly Func<WellKnownSynchronizationKind, string> s_logKind = k => k.ToString();
 
-    private readonly SolutionServices _workspaceServices;
-
-    private readonly Lazy<TemporaryStorageService> _storageService;
-    private readonly ITextFactoryService _textService;
-    private readonly IDocumentationProviderService? _documentationService;
-    private readonly IAnalyzerAssemblyLoaderProvider _analyzerLoaderProvider;
+    // Serialization to temporary storage is only involved when we have a remote process.  Which is only in VS. So the
+    // type of the storage service here is well known.  However the serializer is created in other cases (e.g. to
+    // compute project state checksums). So lazily instantiate the storage service to avoid attempting to get the
+    // TemporaryStorageService when not available.
 
```

Comment:
This blank line should probably be omitted (the preceding comment is specific to one field).